source/causal-consistency/causal-consistency.md

```diff
@@ -0,0 +1,422 @@
+    the `ReadConcern` value and the `afterClusterTime` field.
+    - skip this test if connected to a deployment that does not support cluster times
+    - `session = client.startSession(causalConsistency = true)`
+    - configure collection to use a custom ReadConcern
+    - `collection.findOne(session, {})`
+    - `operationTime = session.operationTime`
+    - `collection.anyReadOperation(session, ...)`
+    - capture the command sent to the server (using APM or other mechanism)
+    - assert that the command has a `level` field with a value matching the custom readConcern
+    - assert that the command has an `afterClusterTime` field with a value of `operationTime`
+10. **Removed**
+11. When connected to a deployment that does not support cluster times messages sent to the server should not include
+    `$clusterTime`.
+    - skip this test when connected to a deployment that does support cluster times
+    - `document = collection.findOne({})`
+    - capture the command sent to the server
+    - assert that the command does not include a `$clusterTime` field
+12. When connected to a deployment that does support cluster times messages sent to the server should include
+    `$clusterTime`.
+    - skip this test when connected to a deployment that does not support cluster times
+    - `document = collection.findOne({})`
+    - capture the command sent to the server
+    - assert that the command includes a `$clusterTime` field
+
+## Motivation
+
+To support causal consistency. Only supported with server version 3.6 or newer.
+
+## Design Rationale
+
+The goal is to modify the driver API as little as possible so that existing programs that don't need causal consistency
+don't have to be changed. This goal is met by defining a `SessionOptions` field that applications use to start a
+`ClientSession` that can be used for causal consistency. Any operations performed with such a session are then causally
+consistent.
+
+The `operationTime` is tracked on a per `ClientSession` basis. This allows each `ClientSession` to have an
+`operationTime` that is sufficiently new to guarantee causal consistency for that session, but no newer. Using an
+`operationTime` that is newer than necessary can cause reads to block longer than necessary when sent to a lagging
+secondary. The goal is to block for just long enough to guarantee causal consistency and no longer.
+
+## Backwards Compatibility
+
+The API changes to support sessions extend the existing API but do not introduce any backward breaking changes. Existing
+programs that don't use causal consistency continue to compile and run correctly.
+
+## Reference Implementation
+
+A reference implementation must be completed before any spec is given status "Final", but it need not be completed
+before the spec is “Accepted”. While there is merit to the approach of reaching consensus on the specification and
```

Comment:
Out-of-scope nit: if possible, I'd be in favor of a lint check to avoid curly quotes in spec files.

Reply:
I'm not aware of any such linter.  :).  I replaced them manually in this file.